src/Models/AZ3166Device.ts

```diff
@@ -168,7 +172,21 @@ export class AZ3166Device extends ArduinoDeviceBase {
       return retValue;
     } else if (configSelection.detail === "Config Connection String") {
       // Get IoT Hub device connection string from config
-      let deviceConnectionString = ConfigHandler.get<string>(ConfigKey.iotHubDeviceConnectionString);
+      const componentConfigs = await this.azureConfigFileHandler.getSortedComponents(ScaffoldType.Workspace);
+      let deviceConnectionString: string | undefined;
+
+      if (componentConfigs.length > 0) {
+        // check if there are any IOT Hub Device Components
+        for (const componentConfig of componentConfigs) {
+          if (componentConfig.type === ComponentType.IoTHubDevice) {
+            if (componentConfig.componentInfo) {
+              deviceConnectionString = componentConfig.componentInfo.values.iotHubDeviceConnectionString;
+              break;
+            }
+          }
+        }
+      }

```

Comment:
This part of logic appears multiple times. Suggest move to a function to both reduce replicated code and increase code clarity.

Reply:
Accept. It has better way to do that.

Reply:
It seem this could be made as a method of AzureConfigFileHandler which returning config value with key under specified component. This would be more reusable than directly manipulate the config file content.

Reply:
Added a new method in AzureConfigFileHandler named "getComponentByTYpe".